Filter box view component

matiasmolleja opened the pull request:
I'm trying to improve the filter box. 
Fixing some issues and trying to make it reusable on other pages as suggested #2037 .
Added "Owned by me" feature. Added sort direction (asc, desc).
This PR is a first step. I just wanted to go step by step to make it easy to merge and see if the path I'm following is OK.

Fixes #2101
Fixes #2087
Fixes #2123

sebastienros commented:
Did you follow the same UI for filters as VSTS?

sebastienros commented:
Can you share a gif on this PR?

matiasmolleja commented:
I did not worked yet on adapt to what VSTS do, but I will work on that next. 
Just wanted to separate the thing to its own component and fix a few things before.

Here is a gif:

![20180712_223234](https://user-images.githubusercontent.com/2589629/42657988-45ceae30-8624-11e8-816e-d987035e6c57.gif)

sebastienros commented on src/OrchardCore.Modules/OrchardCore.Contents/Services/IQueryFromFilterBox.cs:
```diff
@@ -0,0 +1,17 @@
+
+namespace OrchardCore.Contents.Services
+{
+    public interface IQueryFromFilterBox
```
I don't think we need an interface, we can directly register and resolve the class, no need for an abstraction. Even more if this is not in the abstractions project.

Same for ISyerContentTypesProvider. Just merge the two classes into one.

sebastienros replied:
You resolve the class itself, like you were doing before with the interface.

sebastienros commented:
Can you explain why you are doing this refactoring though?
Why would we need to reuse this component? Even more if we know we need to work on the Content Tree at some point which will replace this. We pulled the filter because it was a simple solution that was good enough for the time begin.

matiasmolleja commented:
It started working on it after this discussion: #2037

The idea was to made something that can be reused on other pages. I imagine for example a content item picker, or any other page where we need to show a list of content items. If we have a filterboxViewComponent and a ContentItemListViewComponent we could use them combined anywhere (although to be honest I did not fully validated this approach yet, and I'm not sure if it will be easy to reuse this thing) .

I wanted too to style the content list page to make it close to VSTS, as I recently did for the assets module. After pushing this PR I made pretty good progress on that and now in my local copy I have a nice looking list page.

When I started I saw a few issues and fixed them too ( #2087 , #2101 ).
If you think it is better I can stop working on this to avoid duplicating efforts with the content tree development.

matiasmolleja commented:
I added the changes I mentioned about adding the new fluent-like style.
I added too a submit on change feature to avoid having to click on apply on each change (#2123)
There are still many thing that can be improved but maybe it's better to wait for the content-tree changes.
In the last commit I'm removing a couple of templates that are not needed after my changes. Let me know if it is ok to remove them.

![content-list-fluent](https://user-images.githubusercontent.com/2589629/43411124-9476a296-9429-11e8-9573-9bf00c89de7a.gif)

matiasmolleja commented:
Regarding the published-draft labels, I tried different options. I'm completely sure sure about what version do I prefer. We can change. Here are the options I have:

![status-options](https://user-images.githubusercontent.com/2589629/43411800-8c311196-942b-11e8-8ab2-72069159b811.jpg)

sebastienros commented:
Looks nice, my own comments:
- I don't like the lower-casing of the headers
- You could use a single icon (in the green and blue buttons) and the text as tootlip. Header would just be "Status"
- I prefer to see the `one day ago` text instead of full dates, which can go in tooltips instead
- You just need to show Created. The modified date can be in the draft icon tooltip and the published on the published icon.

You are the designated owner of the Content Tree now ;) Congrats !

matiasmolleja commented:
After trying the last changes, I think it's weird having a selector to sort by modified date, or published date, and not having the corresponding columns on the table. The user will switch for example to modified date, ascending, and he won't have a way to verify that the sorting has been applied (other than start hovering over the icons in sequence).
If the reason to hide the columns is unclutter the view, maybe it's better to let the user show/hide colums (as VSTS do). That can be quite easy to do on the client.

sebastienros commented:
I am closing this PR because we need some better reasons to have this component first. And it's old and not active. I am not deleting the branch for now as I assume you would not like to lose the code.
